server/src/main/java/io/unitycatalog/server/service/AuthService.java

```diff
@@ -167,17 +168,37 @@ public HttpResponse grantToken(
             String cookieTimeout =
                 ServerProperties.getInstance().getProperty("server.cookie-timeout", "P5D");
             Cookie cookie =
-                Cookie.secureBuilder(AuthDecorator.UC_TOKEN_KEY, accessToken)
-                    .path("/")
-                    .maxAge(Duration.parse(cookieTimeout).getSeconds())
-                    .build();
+                createCookie(
+                    AuthDecorator.UC_TOKEN_KEY,
+                    accessToken,
+                    "/",
+                    Duration.parse(cookieTimeout).getSeconds());
             responseHeaders.add(HttpHeaderNames.SET_COOKIE, cookie.toSetCookieHeader());
           }
         });
 
     return HttpResponse.ofJson(responseHeaders.build(), response);
   }
 
+  @Get("/logout")
+  public HttpResponse logout(HttpRequest request) {
+    String authorizationCookie =
+        request.headers().cookies().stream()
+            .filter(c -> c.name().equals(AuthDecorator.UC_TOKEN_KEY))
+            .map(Cookie::name)
+            .findFirst()
+            .orElse(null);
```

Comment:
Why is this needed since you're doing `if` afterwards?

What do you think of getting rid of `orElse(null)` and chaining further to avoid `if` (that should really be called when there's non-`null` value and hence when `orElse` would not be needed)? It makes a lot of sense, doesn't it?

Reply:
Yeah, seems reasonable to just chain and either generate a `ResponseHeaders` with a new cookie, or an empty `ResponseHeaders`. Something like 
```
     request.headers().cookies().stream()
            .filter(c -> c.name().equals(AuthDecorator.UC_TOKEN_KEY))
            .findFirst()
            .map(ResponseHeaders.of(createCookie(...))
            .orElse(ResponseHeaders.of());
```